compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -5634,6 +5744,9 @@ boolean isNeverTypeOrStructureTypeWithARequiredNeverMember(BType type, Set<BType
                 visitedTypeSet.add(arrayType.eType);
                 return arrayType.state != BArrayState.OPEN &&
                         isNeverTypeOrStructureTypeWithARequiredNeverMember(arrayType.eType, visitedTypeSet);
+            case TypeTags.TYPEREFDESC:
+                return isNeverTypeOrStructureTypeWithARequiredNeverMember(getReferredType(type),
```

Comment:
There is a stack overflow issue here.
```Ballerina
type CT record {
    ET i;
};

type DT CT;
type ET DT;

public function main() {
    CT t = {
        i : 2
    };
}
```


```LogTalk
 java.lang.StackOverflowError
        at java.base/java.util.LinkedHashMap$LinkedValues.iterator(LinkedHashMap.java:596)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isNeverTypeOrStructureTypeWithARequiredNeverMember(Types.java:5720)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isNeverTypeOrStructureTypeWithARequiredNeverMember(Types.java:5748)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isNeverTypeOrStructureTypeWithARequiredNeverMember(Types.java:5725)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isNeverTypeOrStructureTypeWithARequiredNeverMember(Types.java:5748)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isNeverTypeOrStructureTypeWithARequiredNeverMember(Types.java:5725)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isNeverTypeOrStructureTypeWithARequiredNeverMember(Types.java:5748)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.isNeverTypeOrStructureTypeWithARequiredNeverMember(Types.java:5725)
```

Reply:
Fixed and added a test 03afd8fa657e179e6278dbe2cf7adad90b542c89